lib/htmlbars-minifier.js

```diff
@@ -1,18 +1,25 @@
-const Filter = require('broccoli-filter');
+const Filter = require('broccoli-persistent-filter');
+const md5Hex = require('broccoli-persistent-filter/lib/md5-hex');
 
-function HtmlbarsMinifier(inputTree, options) {
+function HtmlbarsMinifier(inputNode, options) {
 	if (!(this instanceof HtmlbarsMinifier)) {
-		return new HtmlbarsMinifier(inputTree, options);
+		return new HtmlbarsMinifier(inputNode, options);
 	}
-	this.options = options || {};
 
-	Filter.call(this, inputTree);
+	this.options = Object.assign(options || {}, { persist: true });
+
+	Filter.call(this, inputNode);
 }
 
 HtmlbarsMinifier.prototype = Object.create(Filter.prototype);
 HtmlbarsMinifier.prototype.constructor = HtmlbarsMinifier;
 HtmlbarsMinifier.prototype.extensions = ['hbs'];
 HtmlbarsMinifier.prototype.targetExtension = 'hbs';
+
+HtmlbarsMinifier.prototype.cacheKeyProcessString = function(string, relativePath) {
+	return md5Hex(string.length + 0x00 + relativePath);
```

Comment:
Using a cache key with the length of the content in a plain text file can be wrong.
If I have a file with the work `foo` and I change its content to `bar`, the length is the same and the `cacheKey` will not change.

You must use the file content to generate the cache key:

```js
return md5Hex(string + 0x00 + relativePath);
```

Reply:
done